File: scripts/create-superadmin.ts
import { hashPassword } from "../lib/auth"
import { sql } from "../lib/db"

async function createSuperAdmin() {
  const email = "spencer@miami.ai"
  const password = "PAssword!33!"
  const name = "Spencer"
  const role = "owner"

  try {
    // Check if user already exists
    const existing = await sql`SELECT id FROM users WHERE email = ${email}`

    if (existing.length > 0) {
      console.log("User already exists:", email)
      // Update to owner role if needed
      await sql`UPDATE users SET role = ${role} WHERE email = ${email}`
      console.log("Updated user role to owner")
      return
    }

    // Hash the password
    const passwordHash = await hashPassword(password)
    const userId = crypto.randomUUID()

    // Create the superadmin user
    await sql`
      INSERT INTO users (id, email, password_hash, name, role, created_at, updated_at)
      VALUES (${userId}, ${email}, ${passwordHash}, ${name}, ${role}, NOW(), NOW())
    `

    console.log("✓ Superadmin user created successfully!")
    console.log("Email:", email)
    console.log("Role:", role)
    console.log("User ID:", userId)
  } catch (error) {
    console.error("Failed to create superadmin user:", error)
    throw error
  }
}

createSuperAdmin()

File: scripts/create-superadmin.py
import hashlib
import secrets
import asyncio

# Password to hash
password = "PAssword!33!"

# Generate random 16-byte salt
salt = secrets.token_hex(16)

# Hash password with salt using SHA-256
password_with_salt = password + salt
hash_hex = hashlib.sha256(password_with_salt.encode()).hexdigest()

# Combine salt and hash
password_hash = f"{salt}:{hash_hex}"

print("Generated password hash:")
print(password_hash)
print("\nSQL to insert superadmin user:")
print(f"""
INSERT INTO users (id, email, password_hash, name, role, created_at, updated_at)
VALUES (gen_random_uuid(), 'spencer@miami.ai', '{password_hash}', 'Spencer', 'owner', NOW(), NOW())
ON CONFLICT (email) DO UPDATE 
SET password_hash = EXCLUDED.password_hash, role = 'owner';
""")
